src/plugins/intel_cpu/src/nodes/scatter_update.cpp

```diff
@@ -56,47 +56,147 @@ ScatterUpdate::ScatterUpdate(const std::shared_ptr<ov::Node>& op, const GraphCon
+                break;
+            case OpReduction::MIN:
+                reduction_type = ScatterUpdate::Reduction::MIN;
+                break;
+            case OpReduction::NONE:
+                reduction_type = ScatterUpdate::Reduction::NONE;
+                break;
+            default:
+                THROW_CPU_NODE_ERR("ScatterNDUpdate CPU does not support reduction mode: ", ov::as_string(node_element->get_reduction()));
+        }
     }
 }
 
+void ScatterUpdate::getSupportedDescriptors() {
+    if ((getParentEdges().size() != 3) && (getParentEdges().size() != 4))
+        THROW_CPU_NODE_ERR(errorPrefix, " has incorrect number of input edges");
+    if (getChildEdges().empty())
+        THROW_CPU_NODE_ERR(errorPrefix, " has incorrect number of output edges");
+}
+
 void ScatterUpdate::initSupportedPrimitiveDescriptors() {
     if (!supportedPrimitiveDescriptors.empty())
         return;
+    indicesPrec = getOriginalInputPrecisionAtPort(INDICES_ID);
+    auto indicesType = DnnlExtensionUtils::ElementTypeToDataType(indicesPrec);
+    indicesSize = DnnlExtensionUtils::sizeOfDataType(indicesType);
+    if (indicesSize >= 8) {
+        indicesPrec = ov::element::i64;
+        indicesSize = 8;
+    } else {
+        indicesPrec = ov::element::i32;
+        indicesSize = 4;
+    }
 
+    if (axisRelaxed) {
+        axisPrec = getOriginalInputPrecisionAtPort(AXIS_ID);
+        auto axisType = DnnlExtensionUtils::ElementTypeToDataType(axisPrec);
+        axisSize = DnnlExtensionUtils::sizeOfDataType(axisType);
+        if (axisSize >= 8) {
+            axisPrec = ov::element::i64;
+            axisSize = 8;
+        } else {
+            axisPrec = ov::element::i32;
+            axisSize = 4;
+        }
+    }
+
+    dataPrec = getOriginalInputPrecisionAtPort(DATA_ID);
+    if (one_of(scatterUpdateMode, ScatterUpdateMode::ScatterElementsUpdate, ScatterUpdateMode::ScatterNDUpdate) &&
+        !one_of(dataPrec,
+                ov::element::f32,
+                ov::element::i32,
+                ov::element::bf16,
+                ov::element::f16,
+                ov::element::u8,
+                ov::element::i8)) {
+        dataPrec = ov::element::f32;
+    }
+    dataSize = dataPrec.size();
+
+    bool canBeInplace = !getParentEdgeAt(DATA_ID)->getParent()->isConstant();
+
+    std::vector<PortConfigurator> inPortConfig{{LayoutType::ncsp, dataPrec, false, canBeInplace ? 0 : -1},
+                                               {LayoutType::ncsp, indicesPrec},
+                                               {LayoutType::ncsp, dataPrec}};
+    if (axisRelaxed)
+        inPortConfig.emplace_back(LayoutType::ncsp, axisPrec);
+    addSupportedPrimDesc(inPortConfig,
+                         {{LayoutType::ncsp, dataPrec, false, canBeInplace ? 0 : -1}},
+                          impl_desc_type::unknown);
+}
+
+bool ScatterUpdate::needPrepareParams() const {
+    return true;
+}
+
+void ScatterUpdate::prepareParams() {
```

Comment:
PrepareParams is called on each inference which introduce some performance overhead. Is there any reason why you moved the code below from `initSupportedPrimitiveDescriptors` to `initSupportedPrimitiveDescriptors`?

Reply:
Thanks, sorry, probably I misunderstood one of previous reviews about moving shape/rank checks to constructor/prepareParams, will move it back.
However I wonder, given those checks are just for shape/rank, shouldn't those checks be deleted in favor of core `validate_and_infer_types`?

Reply:
Yeah, if the same checks are presented in shape inference impl, than we could delete them from CPU node. I believe these conditions are more or less legacy

Reply:
I've reverted those changes and deleted prepareParams. Will create leftover ticket to review core shape validation of scatter ops to check if it could be removed from CPU